### ISSUE TITLE
Missing NCL script

### ISSUE THREAD

**Issue by gunnartl** [ESCOMP/CTSM#1380]
In the [documentation on generating CTSM surface datasets for WRF ](https://escomp.github.io/ctsm-docs/versions/master/html/lilac/specific-atm-models/wrf-tools.html) there is a reference to "create_scrip_file.ncl", which is nowhere to be found.

**Comment by kjetilaas** [comment]
Hi! Thanks for raising this issue, @gunnartl. 

Just wanted to add that we are a couple of people here in Oslo who have looked at this together, and cannot get past the point of creating CTSM surface dataset (where this script is needed). Also, we do not have access to cheyenne, where some of these input files might already be available. 
 
There is a mention of this missing ncl script in a "Todo" at the bottom of the documentation, which I think is mentioning @slevisconsulting (although I could be wrong).

**Comment by kjetilaas** [comment]
Great! thanks a lot @negin513

**Comment by dlawrenncar** [comment]
@kjetlaas Please let us know if you have any further problems creating the
surface datasets and we can help.  We are also working towards a new
surface dataset generation workflow that should be available soon.  Would
be great, once that is ready if you could help us test it as a friendly
user.

**Comment by kjetilaas** [comment]
Thanks, @dlawrenncar! We will.

**Comment by negin513** [comment]
Okay @kjetilaas and @gunnartl, I just put the `create_scrip_file.ncl` under `tools/contrib/`.
You can just do a `git pull origin master` to receive the latest changes on master on your local repository. 

Alternatively, you can find the file [here](https://github.com/ESCOMP/CTSM/blob/master/tools/contrib/create_scrip_file.ncl).

As you are going through the steps, you might also find other issues. Please let us know if there are any further problems/issues.

**Comment by negin513** [comment]
Before closing this issue, I'd like to modify the documentation to point where `create_scrip_file.ncl` is located. I am adding it as a to-do item:

- [ ] add clarifications to ctsm-docs on where `create_scrip_file.ncl` script is located.

**Comment by irismuz** [comment]
@negin513 Thank you for uploading `create_scrip_file.ncl`. 
The land output from `create_scrip_file.ncl`, `grid_imask` variable of `wrf2clm_land.nc`, does not contain only ones.
`mkunitymap.ncl` provides the following error: `(0) ERROR: the mask of the second file isn't identically 1!` Could you please double-check `create_scrip_file.ncl`?

**Comment by negin513** [comment]
@irismuz I've just tested the `create_scrip_file.ncl` and it seems like it is working for the `geo_em.d01.nc` file that I am using.

Are you pointing to the correct geo_em output file in your ncl code?

In the ncl code, you should modify this line to point to the correct file:
```
wrf_file  = addfile("geo_em.d01.nc", "r")
```

-----
For skipping the error that you are receiving in `mkunitymap.ncl`, please removing the following lines from `mkunitymap.ncl` and rerun it:

https://github.com/ESCOMP/CTSM/blob/1e3cc271df6b452d25f9115856cbb6218215c036/tools/mkmapdata/mkunitymap.ncl#L77 

This will still throw out the error/warning but will create the `wrf2clm_mapping_noneg.nc` that is needed for the next steps.

**Comment by irismuz** [comment]
@negin513 thank you for checking. We have adhered to your input and the documentation, however, it seems that the `geo_em.d01.nc` file which we use causes that the land output from `create_scrip_file.ncl` in our case does not only contain ones.

**Comment by negin513** [comment]
> @negin513 thank you for checking. We have adhered to your input and the documentation, however, it seems that the `geo_em.d01.nc` file which we use causes that the land output from `create_scrip_file.ncl` in our case does not only contain ones.

Yes. I agree. I think the `geo_em.d01.nc` file that you are using is causing this issue; however, if you'd like to resolve the issue you can simply delete the `exit` line (I mentioned above) in `mkunitymap.ncl`. 
If you remove that line, it should create the weight file that you need without any problems.

**Comment by irismuz** [comment]
@negin513 thank you for your input. Removing line 77 in `mkunitymap.ncl` did not work for us. We think that the issue is specific to our `geo_em.d01.nc`.

**Comment by billsacks** [comment]
It looks like this was mostly done, so closing.
